tests/suites/test_suite_psa_crypto.data

```diff
@@ -3611,9 +3643,57 @@ PSA AEAD setup: invalid algorithm (ChaCha20 - Poly1305 with short tag)
 depends_on:PSA_WANT_ALG_CHACHA20_POLY1305:PSA_WANT_KEY_TYPE_CHACHA20
 aead_multipart_setup:PSA_KEY_TYPE_CHACHA20:"808182838485868788898a8b8c8d8e8f909192939495969798999a9b9c9d9e9f":PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CHACHA20_POLY1305,12):PSA_ERROR_NOT_SUPPORTED
 
-PSA AEAD setup: AES - CCM, invalid tag length 18
+PSA AEAD setup: AES - CCM, invalid tag length 0
+depends_on:PSA_WANT_ALG_CCM:PSA_WANT_KEY_TYPE_AES
+aead_multipart_setup:PSA_KEY_TYPE_AES:"4189351B5CAEA375A0299E81C621BF43":PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CCM,0):PSA_ERROR_INVALID_ARGUMENT
+
+PSA AEAD setup: AES - CCM, invalid tag length 2
+depends_on:PSA_WANT_ALG_CCM:PSA_WANT_KEY_TYPE_AES
+aead_multipart_setup:PSA_KEY_TYPE_AES:"4189351B5CAEA375A0299E81C621BF43":PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CCM,2):PSA_ERROR_INVALID_ARGUMENT
+
+PSA AEAD setup: AES - CCM, invalid tag length 3
 depends_on:PSA_WANT_ALG_CCM:PSA_WANT_KEY_TYPE_AES
-aead_multipart_setup:PSA_KEY_TYPE_AES:"4189351B5CAEA375A0299E81C621BF43":PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CCM,18):PSA_ERROR_INVALID_ARGUMENT
+aead_multipart_setup:PSA_KEY_TYPE_AES:"4189351B5CAEA375A0299E81C621BF43":PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CCM,3):PSA_ERROR_INVALID_ARGUMENT
+
+PSA AEAD setup: AES - CCM, invalid tag length 15
+depends_on:PSA_WANT_ALG_CCM:PSA_WANT_KEY_TYPE_AES
+aead_multipart_setup:PSA_KEY_TYPE_AES:"4189351B5CAEA375A0299E81C621BF43":PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CCM,15):PSA_ERROR_INVALID_ARGUMENT
+
+PSA AEAD setup: AES - CCM, invalid tag length 17
+depends_on:PSA_WANT_ALG_CCM:PSA_WANT_KEY_TYPE_AES
+aead_multipart_setup:PSA_KEY_TYPE_AES:"4189351B5CAEA375A0299E81C621BF43":PSA_ALG_AEAD_WITH_SHORTENED_TAG(PSA_ALG_CCM,17):PSA_ERROR_INVALID_ARGUMENT
+
+PSA AEAD setup: AES - GCM, invalid tag length 0
```

Comment:
Correct values for GCM: `4, 8, 12, 13, 14, 15, 16`.
We have here the following sections for negative test cases and tested values:
```
1. < 4      : 0, 2, 3
2. <5, 7>   : 
3. <9, 11>  : 11
4. > 16     : 17
```

I think we should test at least the boundary values of each section. So my suggestion is to test the following values:
```
0, 3
5, 7
9, 11
17
```

I think the same approach should be for `CCM` and `ChaCha20-Poly1305`.

Reply:
That's one of the possible approaches, but you can also see that GCM accepts even numbers apart from `2` and `10`, that's why I added a test for `2`. Now adding for 10 too. As for CCM... if you want to cover every transition boundary, then we'd need 0, 3, 5, 7, 9, 11, 13, 15, 17... That's a bit too much coverage in my opinion, as we can just check the basic principle (even vs odd). As for ChaCha - it accepts only 16, so 0, 15, 17 is fine.

Reply:
Yes I know that this is a lot of test cases, but usually checking boundary values is the minimum required. 

> ChaCha - it accepts only 16, so 0, 15, 17 is fine.
Yes, if only 16 is valid, then to cover boundary values for negative cases we need `0, 15, 17`.

Maybe someone else should give input here.

Reply:
@mprse required by what? If that is a formal strong requirement we should do it as defined, regarldess of the overhead. If that is an implied requirement, then we could skip some. I am not very familiar with how much time this adds to testing, but if its minimal I would go for better coverage.

If we are looking to optimize the testing for CCM the valid inputs are `4, 6, 8, 10, 12, 14, 16` and the check condition is 

`if( tag_len < 4 || tag_len > 16 || tag_len%2 )`, it would make sense to check:

0 --> ( true || false || true )
2 --> ( true || false || true )
3 --> ( true || false || false )
5 --> ( false || false|| false ) . This is the equivalent for any of the `5, 7, 9, 11, 13, 15`
17 --> ( false || true|| false )
18 --> ( false || true|| true)

Anything from the accepted numbers covers the (false, false, true) case

I have no strong prefference on the testing choice, so overall the patch looks good to me.

Reply:
I also don't have strong preferences on testing and  I don't know what are the testing requirements in this project. 
From my experience and maybe this is not a best option here (safety-critical software) we were testing at least the boundary values.

Reply:
There doesn't seem to be much computation before we exit each negative test case - I'll add more CCM cases.

Reply:
Done :)